api/src/main/java/io/druid/data/input/impl/FileIteratingFirehose.java

```diff
@@ -44,13 +47,26 @@ public FileIteratingFirehose(
   {
     this.lineIterators = lineIterators;
     this.parser = parser;
+    final ParseSpec parseSpec = parser.getParseSpec();
+    if (parseSpec instanceof CSVParseSpec) {
```

Comment:
I thought about signalling with nulls and it sounds good because it is more extendable so users can decide which rows should be skipped. However, i think it's difficult to avoid instanceof because we need to check the skipHeaderRows option is used for non-Hadoop batch index tasks or not.